posthog/warehouse/models/table.py

```diff
@@ -157,32 +159,42 @@ def get_columns(
         self,
         safe_expose_ch_error: bool = True,
     ) -> DataWarehouseTableColumns:
+        placeholder_context = HogQLContext(team_id=self.team.pk)
+        s3_table_func = build_function_call(
+            url=self.url_pattern,
+            format=self.format,
+            access_key=self.credential.access_key,
+            access_secret=self.credential.access_secret,
+            context=placeholder_context,
+        )
         try:
-            placeholder_context = HogQLContext(team_id=self.team.pk)
-            s3_table_func = build_function_call(
-                url=self.url_pattern,
-                format=self.format,
-                access_key=self.credential.access_key,
-                access_secret=self.credential.access_secret,
-                context=placeholder_context,
-            )
-
-            tag_queries(team_id=str(self.team.pk), table_id=str(self.id), warehouse_query=True)
-
-            result = sync_execute(
-                f"""DESCRIBE TABLE (
-                    SELECT *
-                    FROM {s3_table_func}
-                    LIMIT 1
-                )""",
-                args=placeholder_context.values,
-            )
-        except Exception as err:
-            capture_exception(err)
-            if safe_expose_ch_error:
-                self._safe_expose_ch_error(err)
-            else:
-                raise
+            quoted_placeholders = {k: f"'{v}'" for k, v in placeholder_context.values.items()}
+            # chdb doesn't support parameterized queries
+            chdb_query = f"DESCRIBE TABLE (SELECT * FROM {s3_table_func} LIMIT 1)" % quoted_placeholders
+
+            chdb_result = chdb.query(chdb_query, output_format="CSV")
```

Comment:
Why output instead of arrow? I am not super familiar with chdb to be honest, but it looks like they support it.

Reply:
Do we plan to extend this for all warehouse queries? I think a large chunk of our queries work with s3 files rather than clickhouse storage and could be moved over.

Reply:
Nah, just the import queries that don't require any other data and are small in complexity